Example-Crypto-REXX/RXCRYPT1.rexx

```diff
@@ -0,0 +1,2453 @@
+ /**************************************************************/
+ CPCs = JSON_getArrayDim(CPCArray)
+ if CPCs <= 0 then
+    return fatalError( '** Unable to retrieve number of CPCs **' )
+
+ /********************************************************************/
+ /* Traverse the CPCs array to populate CPCsList.                    */
+ /* We use the REXX (1-based) idiom  but adjust for the differing    */
+ /* (0-based) idiom of the toolkit.                                  */
+ /********************************************************************/
+ say 'Processing information for '||CPCs||' CPC(s)... searching for LOCAL'
+ do i = 1 to CPCs
+    nextEntryHandle = JSON_getArrayEntry(CPCArray,i-1)
+    CPClocation = JSON_findValue(nextEntryHandle,"location",HWTJ_STRING_TYPE)
+
+    if CPClocation = 'local' then
+      do
+        say 'found LOCAL CPC'
+        if localCPCfound = TRUE then
+          do
+            return fatalError('** found two LOCAL CPCs **')
+          end
+        localCPCfound = TRUE
+
+      CPCversion=JSON_findValue(nextEntryHandle,"se-version",HWTJ_STRING_TYPE)
+        if foundJSONValue(CPCversion) = FALSE then
+          do
+            return fatalError,
+            ('ERROR: found LOCAL CPC but it did not contain an se-version')
+          end
+
+        CPCuri=JSON_findValue(nextEntryHandle,"object-uri",HWTJ_STRING_TYPE)
+        if foundJSONValue(CPCuri) = FALSE then
+          do
+            return fatalError,
+            ('ERROR: found LOCAL CPC but it did not contain an object-uri')
+          end
+
+        CPCtargetName=JSON_findValue(nextEntryHandle,"target-name",,
+                                    HWTJ_STRING_TYPE)
+        if foundJSONValue(CPCtargetName) = FALSE then
+          do
+            return fatalError,
+            ('ERROR: found LOCAL CPC but it did not contain target-name')
+          end
+
+        CPCname=JSON_findValue(nextEntryHandle,"name",HWTJ_STRING_TYPE)
+        if foundJSONValue(CPCname) = FALSE then
+          do
+            return fatalError,
+            ('ERROR: found LOCAL CPC but it did not contain a name property')
+          end
+      end
+ end /* endloop thru the JSON LPARs array */
+
+if localCPCfound then
+  do
+    Say
+    Say 'Successfully obtained LOCAL CPC Info:'
+    Say '  name:'||CPCname
+    Say '  uri:'||CPCuri
+    Say '  target-name:'||CPCtargetName
+    Say '  SE version: '||CPCversion
+    Say                                                                 
+    if substr(CPCversion,3,2) < 16 then                                 
+      do                                                                
+        errMsg = '** Crypto attributes supported on z16 or higher only **'
+        return fatalError(errMsg)                                        
+      end                                                               
+      else                                                              
```

Comment:
one less indent for the else so it's the same column as the if, just to make it clear what the else goes with

Reply:
fixed